### PR TITLE
bug all iot components not deleted

### PR DESCRIPTION
**Pull request by famartinrh**
I'm experiencing an issue with  iot tests, deleting the iotconfig resource do not delete all iot components. The annoying thing is that deleting via command line works fine and removes all iot components.

### CODE REVIEW

**Comment by famartinrh** [comment]
@enmasse-ci run tests testcase=**.IoTConfigTest

File: systemtests/src/main/java/io/enmasse/systemtest/utils/IoTUtils.java [hunk 1]
```diff
@@ -58,6 +59,17 @@ public static void waitForIoTConfigReady(Kubernetes kubernetes, IoTConfig config
...
     }
 
+    public static void deleteIoTConfigAndWait(Kubernetes kubernetes, IoTConfig config) throws Exception{
+        String operationID = TimeMeasuringSystem.startOperation(SystemtestsOperation.DELETE_IOT_CONFIG);
+        kubernetes.getIoTConfigClient().withName(config.getMetadata().getName()).delete();
+        waitForIoTConfigDeleted(kubernetes, config);
+        TimeMeasuringSystem.stopOperation(operationID);
+    }
+
+    private static void waitForIoTConfigDeleted(Kubernetes kubernetes, IoTConfig config) throws Exception{
```
Comment (jbtrystram) [inline systemtests/src/main/java/io/enmasse/systemtest/utils/IoTUtils.java]: why passing `IoTConfig config` if it's not used?

Comment (famartinrh) [reply systemtests/src/main/java/io/enmasse/systemtest/utils/IoTUtils.java]: my mistake, but this doesn't affect to the problem

**Comment by famartinrh** [comment]
@enmasse-ci run tests profile=shared-iot

File: systemtests/src/main/java/io/enmasse/systemtest/utils/IoTUtils.java (hunk 1 above)
Comment (jbtrystram) [reply systemtests/src/main/java/io/enmasse/systemtest/utils/IoTUtils.java]: I guess so, I was just wondering :)

**Pull request merged**